dev/AppNotifications/AppNotificationBuilder/AppNotificationBuilderUtility.h

```diff
@@ -70,3 +73,55 @@ inline PCWSTR GetWinSoundEventString(AppNotificationBuilder::AppNotificationSoun
         return L"ms-winsoundevent:Notification.Default";
     }
 }
+
+inline std::wstring Encode(winrt::hstring const& value)
+{
+    std::wstring encodedValue{};
+    for (auto ch : value)
+    {
+        switch (ch) {
+        case '%':
+            encodedValue.append(c_encodedPercent);
+            break;
+        case ';':
+            encodedValue.append(c_encodedSemicolon);
+            break;
+        case '=':
+            encodedValue.append(c_encodedEquals);
+            break;
+        default:
+            encodedValue.push_back(ch);
+        }
+    }
+    return encodedValue;
+}
+
+// This function replaces the encoded values with the actual values
+inline std::wstring DecodeString(std::wstring value, std::wstring const& encodedValue, char ch)
```

Comment:
Can we avoid making a copy of value here?

Reply:
Yeah we can. Need to add another prevPos variable to track where we are in the string. The original idea was to destroy the copied string.

Reply:
Removing in favor of std::regex_replace